liblttng-ust-java-agent/java/lttng-ust-agent-jul/org/lttng/ust/agent/jul/LttngLogHandler.java

```diff
@@ -134,13 +156,17 @@ public void publish(LogRecord record) {
 		 */
 		LttngJulApi.tracepointWithContext(formattedMessage,
 				record.getLoggerName(),
-				record.getSourceClassName(),
-				record.getSourceMethodName(),
+				logSource ? record.getSourceClassName() : EMPTY,
+				logSource ? record.getSourceMethodName() : EMPTY,
 				record.getMillis(),
 				record.getLevel().intValue(),
 				record.getThreadID(),
 				contextInfo.getEntriesArray(),
 				contextInfo.getStringsArray());
 	}
 
+	public void setLogSource(boolean doLog) {
```

Comment:
What is the use case for this method? I'd find it cleaner if `logSource` could be final, and set through properties.

If there is really a need, then you'd have to
* Make logSource `volatile`
* Add Javadoc to this method